c/src/lib/libbsp/arm/shared/stm32fxxxx/spi/hal-spi.c

```diff
@@ -0,0 +1,507 @@
+#endif
+#if (STM32_ENABLE_SPI6)
+ {.port       = SPI6_PORT,
+  .sck_pin    = SPI6_SCK_PIN,
+  .miso_pin    = SPI6_MISO_PIN,
+  .mosi_pin    = SPI6_MOSI_PIN,
+  .alternate_func = GPIO_AF5_SPI6},
+#endif
+};
+
+/**
+ * SPI Bus control
+ */
+stm32_spi_bus *bus[MAX_SPI_INSTANCES];
+
+/**
+ * Returns the base address of the port
+ * TODO:: this function is to be pushed to gpio file and used from there
+ */
+GPIO_TypeDef* stm32_spi_get_gpio_port(gpio_port port)
+{
+   switch(port)
+   {
+   case PORTA:   return GPIOA;
+               break;
+   case PORTB:   return GPIOB;
+               break;
+   case PORTC:   return GPIOC;
+               break;
+   case PORTD:   return GPIOD;
+               break;
+   case PORTE:   return GPIOE;
+               break;
+   }
+}
+
+/**
+ * Returns the base address of the SPI instances
+ */
+SPI_TypeDef* stm32_spi_get_spi_instance(SPI_Instance spi_instance)
+{
+   switch(spi_instance)
+   {
+   case SPI_ONE:   return SPI1;
+               break;
+   case SPI_TWO:   return SPI2;
+               break;
+   case SPI_THREE:   return SPI3;
+               break;
+#if (STM32_ENABLE_SPI4)
+   case SPI_FOUR:   return SPI4;
+               break;
+#endif
+#if (STM32_ENABLE_SPI5)
+   case SPI_FIVE:   return SPI5;
+               break;
+#endif
+#if (STM32_ENABLE_SPI6)
+   case SPI_SIX:   return SPI6;
+               break;
+#endif
+   }
+}
+
+/**
+ * Initialize SPI clock
+ */
+void stm32_spi_initialize_spi_clock(SPI_Instance spi_instance)
+{
+   switch(spi_instance)
+   {
+   case SPI_ONE:   __HAL_RCC_SPI1_CLK_ENABLE();
+               break;
+   case SPI_TWO:   __HAL_RCC_SPI2_CLK_ENABLE();
+               break;
+   case SPI_THREE:   __HAL_RCC_SPI3_CLK_ENABLE();
+               break;
+#if (STM32_ENABLE_SPI4)
+   case SPI_FOUR:   __HAL_RCC_SPI4_CLK_ENABLE();
+               break;
+#endif
+#if (STM32_ENABLE_SPI5)
+   case SPI_FIVE:   __HAL_RCC_SPI5_CLK_ENABLE();
+               break;
+#endif
+#if (STM32_ENABLE_SPI6)
+   case SPI_SIX:   __HAL_RCC_SPI6_CLK_ENABLE();
+               break;
+#endif
+   }
+}
+
+/**
+ * SPI get instance based vector number
+ */
+IRQn_Type stm32_get_spi_vec_num(SPI_Instance spi_instance)
+{
+   switch(spi_instance)
+   {
+   case SPI_ONE:   return (IRQn_Type)GET_SPI_VEC((SPI_ONE++));
+               break;
+   case SPI_TWO:   return (IRQn_Type)GET_SPI_VEC(SPI_TWO++);
+               break;
+   case SPI_THREE:   return (IRQn_Type)GET_SPI_VEC(SPI_THREE++);
+               break;
+   case SPI_FOUR:   return (IRQn_Type)GET_SPI_VEC(SPI_FOUR++);
+               break;
+   case SPI_FIVE:   return (IRQn_Type)GET_SPI_VEC(SPI_FIVE++);
+               break;
+   case SPI_SIX:   return (IRQn_Type)GET_SPI_VEC(SPI_SIX++);
+               break;
+   }
+}
+
+
+/**
+ * SPI IOCTL
+ */
+static int stm32_spi_ioctl(
+  spi_bus *base,
+  uint32_t ioctl_cmd,
+  unsigned long arg
+)
+{
+   rtems_status_code sc = RTEMS_SUCCESSFUL;
+   stm32_spi_bus *bus = (stm32_spi_bus *) base;
+
+   _Assert(base != NULL);
```

Comment:
Also should probably be done before not after line 239. If it is null you should exit before you set bus to base. However, functionally it doesn't matter as the code is currently written, but it is not clear why you are checking base after you already did something with it and if someone adds code before this line as it might not be clear it can cause problems. Remove this possible area of mess up by checking your arguments before you do anything with them to make it a little logically clearer what your intent is